src/BCFSerialize.cc

```diff
@@ -98,12 +113,40 @@ int bcf_raw_read_from_mem(const char *addr, bcf1_t *v) {
     // bcf_subset, prior to and including bd6ed8b4
     if ( (!v->indiv.l || !v->n_sample) && v->n_fmt ) v->n_fmt = 0;
 
-    memcpy(v->shared.s, &addr[loc], v->shared.l);
+    memcpy(v->shared.s, &buf[start + loc], v->shared.l);
     loc += v->shared.l;
-    memcpy(v->indiv.s, &addr[loc], v->indiv.l);
+    memcpy(v->indiv.s, &buf[start + loc], v->indiv.l);
     loc += v->indiv.l;
 
-    return loc;
+    ans = loc;
+    return Status::OK();
+}
+
+// Calculate the total length of a record, without unpacking it.
+Status bcf_raw_calc_rec_len(const char *buf, int start, size_t len, uint32_t &ans) {
+    Status s;
+    S(range_check(start + 32, len));
+
+    uint32_t *x = (uint32_t*) &buf[start];
+    assert(x[0] > 0);
+    ans = 32 + (x[0] -24) + x[1];
+    return Status::OK();
+}
+
+// Check if the record that starts at memory address [addr] overlaps
+// the range [rng]. The trick is to do this without unpacking the record.
+Status bcf_raw_overlap(const char *buf, int start, size_t len, const range &rng,
+                       bool &ans) {
+    ans = false; // extra sanitation
+    Status s;
+    S(range_check(start + 32, len));
+
+    uint32_t *x = (uint32_t*) &buf[start];
+    int32_t rid = x[2];
+    int32_t beg = x[3];
+    int32_t rlen = x[4];
```

Comment:
Why switch between `uint32_t` and `int32_t`? Granted, `range` takes `int`s, but that conversion can be implicit.